tests/cli-rustup.rs

```diff
@@ -226,6 +226,89 @@ fn rustup_no_channels() {
     })
 }
 
+#[test]
+fn check_updates_none() {
+    setup(&|config| {
+        set_current_dist_date(config, "2015-01-01");
+        expect_ok(config, &["rustup", "update", "stable", "--no-self-update"]);
+        expect_ok(config, &["rustup", "update", "beta", "--no-self-update"]);
+        expect_ok(config, &["rustup", "update", "nightly", "--no-self-update"]);
+        expect_stdout_ok(
+            config,
+            &["rustup", "check"],
+            for_host!(
+                r"stable-{0} - Up to date : 1.0.0 (hash-s-1)
+beta-{0} - Up to date : 1.1.0 (hash-b-1)
+nightly-{0} - Up to date : 1.2.0 (hash-n-1)
+"
+            ),
+        );
+    })
+}
+
+#[test]
+fn check_updates_some() {
+    setup(&|config| {
+        set_current_dist_date(config, "2015-01-01");
+        expect_ok(config, &["rustup", "update", "stable", "--no-self-update"]);
+        expect_ok(config, &["rustup", "update", "beta", "--no-self-update"]);
+        expect_ok(config, &["rustup", "update", "nightly", "--no-self-update"]);
+        set_current_dist_date(config, "2015-01-02");
+        expect_stdout_ok(
+            config,
+            &["rustup", "check"],
+            for_host!(
+                r"stable-{0} - Update available : 1.0.0 (hash-s-1) -> 1.1.0 (hash-s-2)
+beta-{0} - Update available : 1.1.0 (hash-b-1) -> 1.2.0 (hash-b-2)
+nightly-{0} - Update available : 1.2.0 (hash-n-1) -> 1.3.0 (hash-n-2)
+"
+            ),
+        );
+    })
+}
+
+#[test]
+fn check_updates_with_update() {
+    setup(&|config| {
+        set_current_dist_date(config, "2015-01-01");
+        expect_ok(config, &["rustup", "update", "stable", "--no-self-update"]);
+        expect_ok(config, &["rustup", "update", "beta", "--no-self-update"]);
+        expect_ok(config, &["rustup", "update", "nightly", "--no-self-update"]);
+        expect_stdout_ok(
+            config,
+            &["rustup", "check"],
+            for_host!(
+                r"stable-{0} - Up to date : 1.0.0 (hash-s-1)
+beta-{0} - Up to date : 1.1.0 (hash-b-1)
+nightly-{0} - Up to date : 1.2.0 (hash-n-1)
+"
+            ),
+        );
+        set_current_dist_date(config, "2015-01-02");
+        expect_stdout_ok(
+            config,
+            &["rustup", "check"],
+            for_host!(
+                r"stable-{0} - Update available : 1.0.0 (hash-s-1) -> 1.1.0 (hash-s-2)
+beta-{0} - Update available : 1.1.0 (hash-b-1) -> 1.2.0 (hash-b-2)
+nightly-{0} - Update available : 1.2.0 (hash-n-1) -> 1.3.0 (hash-n-2)
+"
+            ),
+        );
+        expect_ok(config, &["rustup", "update", "beta", "--no-self-update"]);
+        expect_stdout_ok(
+            config,
+            &["rustup", "check"],
+            for_host!(
+                r"stable-{0} - Update available : 1.0.0 (hash-s-1) -> 1.1.0 (hash-s-2)
+beta-{0} - Up to date : 1.2.0 (hash-b-2)
+nightly-{0} - Update available : 1.2.0 (hash-n-1) -> 1.3.0 (hash-n-2)
+"
+            ),
+        );
+    })
+}
+
```

Comment:
These new tests quite probably belong in `cli-exact` since they're checking very exact formatting outputs.

Reply:
Moved to `cli-exact`